src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/LocationResourceIntTest.kt

```diff
@@ -1312,7 +1351,102 @@ class LocationResourceIntTest : SqsIntegrationTestBase() {
 
-  @DisplayName("GET /locations/prison/{prisonId}/archived")
+  @DisplayName("GET /locations/prison/{prisonId}/groups")
+  @Nested
+  inner class viewLocationGroupsInPropertiesByPrisonTest {
+    @Nested
+    inner class Security {
+
+      @Test
+      fun `access forbidden when no authority`() {
+        webTestClient.get().uri("/locations/prison/MDI/groups")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.get().uri("/locations/prison/MDI/groups")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.get().uri("/locations/prison/MDI/groups")
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
+
+      @Test
+      fun `can retrieve residential groups for a prison included in the properties file`() {
+        webTestClient.get().uri("/locations/prison/MDI/groups")
+          .headers(setAuthorisation(roles = listOf("ROLE_VIEW_LOCATIONS")))
+          .exchange()
+          .expectStatus().isOk
+          .expectBody().json(
+            // language=json
+            """
+                        [{
+                          "name": "Houseblock 1",
+                          "key": "Houseblock 1",
+                          "children": [{
+                            "name": "A-Wing",
+                            "key": "A-Wing",
+                            "children": []
+                          }, 
+                          {},{}
+                          ]},
+                          {
+                          "name": "Houseblock 7",
+                          "key": "Houseblock 7",
+                          "children": []
+                          },
+                          {
+                          "name": "Segregation Unit",
+                          "key": "Segregation Unit",
+                          "children": []
+                          },
+                          {},{},{},{},{},{}  
+                        ]
+                        """,
+            false,
+          )
+      }
+
+      @Test
+      fun `can retrieve residential groups for a prison NOT included in the properties file`() {
+
+        webTestClient.get().uri("/locations/prison/NMI/groups")
+          .headers(setAuthorisation(roles = listOf("ROLE_VIEW_LOCATIONS")))
+          .exchange()
+          .expectStatus().isOk
+          .expectBody().json(
+            // language=json
+            """
+                [{
+                  "name":"Wing A",
+                  "key":"A",
+                  "children":[{"name":"Landing A","key":"A-1","children":[]}]
+                  },
+                  {
+                    "name":"Landing A",
```

Comment:
This is unexpected. 
I expected the child to only be on line 1436 not duplicated at 1439 too